crates/primitives/src/transaction/mod.rs

```diff
@@ -1582,22 +1582,6 @@ impl Decodable for TransactionSignedEcRecovered {
     }
 }
 
-/// Ensures the transaction can be sent over the
-/// network
-pub trait IntoRecoveredTransaction {
-    /// Converts to this type into a [`TransactionSignedEcRecovered`].
-    ///
-    /// Note: this takes `&self` since indented usage is via `Arc<Self>`.
-    fn to_recovered_transaction(&self) -> TransactionSignedEcRecovered;
-}
-
-impl IntoRecoveredTransaction for TransactionSignedEcRecovered {
-    #[inline]
-    fn to_recovered_transaction(&self) -> TransactionSignedEcRecovered {
-        self.clone()
-    }
-}
```

Comment:
@mattsse The main goal of this PR was to remove this implementation which is just a clone and doesn't seem to be used anywhere.

But as I had trouble dealing with `IntoRecoveredTransaction` in general, I took the opportunity to implement a `From` which may be more idiomatic but has the only disadvantage of not being able to be used directly with the `Arc` (we have to go through an intermediate `as_ref`). I know that sometimes you don't like this kind of approach which uses `From` rather than custom traits. Tell me what you think, if necessary I can revert this part relating to the `From` implementation and keep the `IntoRecoveredTransaction` trait around the codebase.